motionplan/plan.go

```diff
@@ -0,0 +1,154 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type Plan struct {
+	Trajectory
+	Path
+
+	// nodes corresponding to inputs can be cached with the Plan for easy conversion back into a form usable by RRT
+	// depending on how the Trajectory is constructed these may be nil and should be computed before usage
```

Comment:
In the case where we build a plan from proto it will not be prepopulated with `nodes`.  Right now this only happens with the round trip tests built around `Plans` but its reasonable to expect that in the future we might have some API method that accepts `Plan` as a parameter.  We would need to walk the `Plan` and the manually construct the nodes that would correspond to the RRT tree

Reply:
Would having a constructor that takes nodes & returns a Plan fix that problem? I think there is a lot of value in a type always being valid (i.e. correct by construction design). I think requiring every part of the code that takes a `Plan` need to consider the case of the `nodes` being nil, when in practice it only happens when the `Plan` is not yet fully instantiated is worse than saying: `nodes is never nil` & having a constructor specifically for creating a plan from nodes.

Reply:
Yes I like the idea of only having plans be constructable with their constructors

Reply:
I think I've addressed this by now